src/vivarium/framework/results/manager.py

```diff
@@ -188,7 +194,12 @@ def gather_results(self, event_name: str, event: Event) -> None:
                     raise ValueError(
                         f"There is a results group {results_group} but no corresponding measure"
                     )
-                self._metrics[measure] += results_group
+                # Look for extra columns in the results_group and initialize with 0.
+                extra_cols = list(set(results_group.columns) - {METRICS_COLUMN})
```

Comment:
Isn't this going to reset the observation to 0 every time, so we won't get any accumulation? We really want to compare the columns in `results_group` to those in `self._metrics[measure]`. If this didn't cause any tests to fail, our coverage likely isn't good enough.